### DISCUSSION TITLE
How can I zoom camera to mouse position in OSG?

### DISCUSSION THREAD

**Discussion by msnh2012** [openscenegraph/OpenSceneGraph#1249] [Q&A]
Like this:
https://m.youtube.com/watch?v=9dU72LKqpJM&pp=ygUbYmxlbmRlciB6b29tIG1vdXNlIHBvc2l0aW9u

**Comment by robertosfield** [comment]
None of the osgGA CamaraManipulators provide this functionality so you will need to implement it yourself, such as by sublcassing from vsgGA::TrackballManipulator.  It's over a decade since I did any work on these classes so I can't recall the methods to override off the top of my head.

**Comment by keince** [comment]
The osgEarth software which uses the OSG under the hood, has a zoom-to-mouse feature in its Earth Manipulator. 

Perhaps you can look at the EarthManipulator.cpp file in osgEarth for inspiration...
